Formula/libxcb.rb

```diff
@@ -1,9 +1,15 @@
+class SystemPython3Requirement < Requirement
+  fatal true
+  satisfy { which "python3" }
```

Comment:
Should we include a slightly more informative error message for this?

I also wonder if we should try to check that `which "python3"` doesn't actually return Homebrew python3... but I suppose given the circular dependency that would be impossible. However, it's not obvious that we'll always have a circular dependency here?

Reply:
I can add the following message:
```
def message
    "System Python 3 is needed to avoid the following cyclic dependency on linux: python3 -> tcl-tk -> libx11 -> libxcb -> python3"
  end
```

Would that be ok for you? That message would be displayed on Mac too though.

I can make the requirement linux-only and let the Mac version depend on python@3.9.

> but I suppose given the circular dependency that would be impossible.

On linux this is probably not be an issue.
On Mac you might install python@3.9, the try to compile libxcb/xcb-proto from source, and you will pick up the brewed python3. But this point it's not a big deal, the build will probably pass, and you are doing something that is not supported on our side.

I used `which "python3"` mostly for linux, as python 3 might be installed god knows where. It's not so important to target a specific python 3 version, anything between 3.9 and 3.5 (and maybe older) will do.

Reply:
I think that's a little _too_ informative. 😅 How about

    An existing non-Homebrew Python 3 installation is required in order to avoid cyclic dependencies!